deploy/docker/start.sh

```diff
@@ -1,47 +1,41 @@
 #!/bin/sh
-set -e
+set -e -u
 
 emqx_exit(){
-    # tail erlang.log.*
-    erlang_log=$(echo $(ls -t /opt/emqx/log/erlang.log.*) | awk '{print $1}')
-    num=$(sed -n -e '/LOGGING STARTED/=' ${erlang_log} | tail -1)
-    tail -n +$((num-2)) ${erlang_log}
+    # tail emqx.log.*
+    num=$(sed -n -e '/LOGGING STARTED/=' /opt/emqx/log/emqx.log | tail -1)
+    tail -n +$((num-2)) /opt/emqx/log/emqx.log
 
     echo "['$(date -u +"%Y-%m-%dT%H:%M:%SZ")']:emqx exit abnormally"
     exit 1
 }
 
 ## EMQ Main script
 
-# Start and run emqx, and when emqx crashed, this container will stop
+# Set log.rotation=off in etc/emqx.conf
+sed -i -r 's/log.rotation = (.*)/log.rotation = off/1' /opt/emqx/etc/emqx.conf
 
+# Start and run emqx, and when emqx crashed, this container will stop
 /opt/emqx/bin/emqx start || emqx_exit
 
-tail -f /opt/emqx/log/erlang.log.1 &
-
 # Sleep 5 seconds to wait for the loaded plugins catch up.
 sleep 5
 
 echo "['$(date -u +"%Y-%m-%dT%H:%M:%SZ")']:emqx start"
 
+tail -f /opt/emqx/log/emqx.log &
```

Comment:
could this fork be a potential leak ? 
https://stackoverflow.com/questions/360201/how-do-i-kill-background-processes-jobs-when-my-shell-script-exits/2173421#2173421

Reply:
since this is docker entrypoint, i guess the risk of leak is low because container would terminate if entrypoint exits.
however I'd argue that the least we can do is to document it as a comment,

Reply:
The "&" symbol is used here because I need to print the log and check the status of emqx. Do you have any better suggestions? I don’t know much about the trap command

Reply:
Examples can be found in the stackoverflow link I shared.
However this is now a minor because container is terminated when parent exits.

The original line 41 however is indeed a leak.

Reply:
I read something about trap, I am not sure if I understand it correctly, `trap "emqx_exit" EXIT` seems to perform the `emqx_exit` operation when the EXIT signal is received

`tail -f /opt/emqx/log/emqx.log &` This line of content exists because I need to print the contents of the log file all the time, rather than just printing at the exit, can the trap do it?

Reply:
emqx_exit is called in two places, line 19 and line 47
I guess in line 47's case, it's not always `erlang.log.1`,
hence the sorting by time and greping by START is still necessary.

> tail -f /opt/emqx/log/emqx.log & This line of content exists because I need to print the contents of the log file all the time, rather than just printing at the exit, can the trap do it?
I meant to say that for printing `erlang.log.X` (the reason why beam exited),
it can be done by trapping EXIT, so there is no need to call the function in two places.
So that when the script exited for some other reason (other than the two calls), it can still print.

For emqx.log, I agree that it makes sense to fork a `tail -f` to have a stream of logs to container log.